### ISSUE TITLE
Replace server author services specific data generators with fake-data-generator

### ISSUE THREAD

**Issue by sgobotta** [Cambalab/vue-admin#161]
In the utils server we are using too many custom functions for the authors fake data. We could replace that using the [fake-data-generator](https://github.com/Cambalab/fake-data-generator) as a dev-dependency.

Current models: please respect the current type of each property

```javascript
author: {
  id: Number, // faker random number
  name: String, // faker random words
  lastname: String, // faker lorem text
  birthdate: ISOString // faker date between
}
```

Checkpoints:

+   [x] ***Run the e2e tests, they should pass.***
+   [x] ***Run the unit tests, they should pass.***
+   [x] **Authors service:** Replace ipsum dependency and the custom date format functions with fake-data-generator. Replace the **authors** array with a new fake-data-generator model of **40 items**
+   [x] ***Rerun e2e tests, they must pass.***
+   [x] ***Rerun unit tests, they must pass.***

**Comment by jphetphoumy** [comment]
Hi @sgobotta, If nobody is working on this issue, I'll be glad to help !

**Comment by sgobotta** [comment]
Hi @jphetphoumy, all yours. Let me know if you need any help of any kind. I think the `fake-data-generator` is ready to accept all the required fields for **authors**

**Comment by jphetphoumy** [comment]
Hi !

I tried to do some test with the fake-data-generator but it does not seems to work as expected.

I tried the following code : 

```js
// Requires the package
const { generateModel } = require('fake-data-generator')
// Requires a model
const model = require('./models/authors.json')
// Generate the model
const amountArg = 50
const modelArg = 'authors'
const outputType = 'object'
const generatedModel = generateModel({ amountArg, modelArg, outputType })
```

if `authors` does not exist in the `node_modules/fake-data-generator/dist/models/`, I have the following error : 

```
 jeremy@ArchLinux  ~/Document/Code/test-fake-data  node index.js                                                                                              SIGINT(2) ↵  10406  14:21:32
internal/modules/cjs/loader.js:797
    throw err;
    ^

Error: Cannot find module '../models/authors'
Require stack:
- /home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js
- /home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/index.js
- /home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/main.js
- /home/jeremy/Document/Code/test-fake-data/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:794:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:849:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at getModel (/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:17:19)
    at createSingle (/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:25:15)
    at /home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:34:12
    at Array.map (<anonymous>)
    at createMany (/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:33:47)
    at exports.default (/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:49:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js',
    '/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/index.js',
    '/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/main.js',
    '/home/jeremy/Document/Code/test-fake-data/index.js'
  ]
}
```

I looked at the `fake-data-generator` code and this [line](https://github.com/Cambalab/fake-data-generator/blob/master/lib/create-model.js#L4) seems to be the problem

So here the question,

Do I need to make a pull request to `fake-data-generator` to add the `authors.json` model or maybe replace 

```js
const modelData = require(`../models/${modelName}`)
```

by 

```js
const modelData = require(path.resolve(`${__dirname}/../models/${modelName}`))
```

to get the current path ?

**Comment by sgobotta** [comment]
Oh, it looks kind of horrible. I didn't know the state of the generator when used as an npm dependency.

I think the solution is a bit more comples than looking for a .json in the right place. Anyhow, I'll open an [issue](https://github.com/Cambalab/fake-data-generator/issues/23) in `fake-data-generator` explaining the situation. Sadly, this task can't be finished until the generator is fixed.

**Comment by sgobotta** [comment]
Alright, I've just released [**v0.1.7**](https://github.com/Cambalab/fake-data-generator/releases) for the `fake-data-generator`. :tada:

**Comment by jphetphoumy** [comment]
Hi ! 

I Have looked into the issue, but the faker date.between doesn't seem to work.

I read the fakerjs documentation and It seems that we need to pass some argument to the faker data.between to generate a date : 

http://marak.github.io/faker.js/faker.date.html

But I think that fake-data-generator can not handle argument yet : 

https://github.com/Cambalab/fake-data-generator/blob/develop/README.md#faker


Also, I didn't see any unit test or e2e test for the authors.js 

Would you like me to write them ?

**Comment by sgobotta** [comment]
> Hi ! 
>
> I Have looked into the issue, but the faker date.between doesn't seem to work.
> 
> I read the fakerjs documentation and It seems that we need to pass some argument to the faker data.between to generate a date : 
> 
> http://marak.github.io/faker.js/faker.date.html
> 
> But I think that fake-data-generator can not handle argument yet : 
> 
> https://github.com/Cambalab/fake-data-generator/blob/develop/README.md#faker


Okay, I've just published `v0.1.8` for fake-data-generator that provides `faker.between` support.
I've made a few manual tests and seems to work fine by passing an array of dates as `options` (see `example.json`).

> Also, I didn't see any unit test or e2e test for the authors.js 
> 
> Would you like me to write them ?

Just e2e tests :) you can find them [here](https://github.com/Cambalab/vue-admin/tree/master/tests/e2e/specs/authors)
